Resource/PYTHON/multilateration/test/test_multilateration.py

```diff
@@ -0,0 +1,186 @@
+                - param.HydrophoneDetails.STARBOARD_HYD_Z,
+                2,
+            )
+        )
+    )
+    dist_src_stern = float(
+        math.sqrt(
+            math.pow(
+                param.TestParameters.SOURCE_POS_X - param.HydrophoneDetails.STERN_HYD_X,
+                2,
+            )
+            + math.pow(
+                param.TestParameters.SOURCE_POS_Y - param.HydrophoneDetails.STERN_HYD_Y,
+                2,
+            )
+            + math.pow(
+                param.TestParameters.SOURCE_POS_Z - param.HydrophoneDetails.STERN_HYD_Z,
+                2,
+            )
+        )
+    )
+
+    # Calculating the numer of samples it takes for the sound to arrive at x hyd
+
+    # Since the TOA uses lag (direct measuremend), these values are uint32_t
+    # converting to int32 works as floor. Should be rounded.
+    samp_port = np.int32(
+        param.DSPConstants.SAMPLE_FREQUENCY
+        * (dist_src_port / param.PhysicalConstants.SOUND_SPEED)
+    )  #
+    samp_starboard = np.int32(
+        param.DSPConstants.SAMPLE_FREQUENCY
+        * (dist_src_starboard / param.PhysicalConstants.SOUND_SPEED)
+    )  #
+    samp_stern = np.int32(
+        param.DSPConstants.SAMPLE_FREQUENCY
+        * (dist_src_stern / param.PhysicalConstants.SOUND_SPEED)
+    )  # *
+
+    lag_port_starboard = samp_port - samp_starboard
+    lag_port_stern = samp_port - samp_stern
+    lag_startboard_stern = samp_starboard - samp_stern
+
+    return [lag_port_starboard, lag_port_stern, lag_startboard_stern]
+
+
+"""
+   * @brief Function that checks if the simulated sound-source position is  within 
+   * @p MARGIN_POS_ESTIMATE distance from the estimated position
+   * 
+   * Writes the distance estimated and the actual distance to the terminal
+"""
+
+
+def test_trilateration_algorithm():
+
+    """
+    Creating the TOA-intervals.
+    Writing out an error-msg if the parameters are wrong
+    """
+
+    lag_array = calculate_toa_array()
+
+    if not valid_number_of_hydrophones:
+        print("\nAt least one parameter is invalid")
+        return
+
+    # Initializing the estimate for x-pos and y-pos
+
+    x_pos_es, y_pos_es = mult.trilaterate_pinger_position(lag_array)
+
+    """
+    * Calculating the distance between the estimated position and the 
+    * actual position. Only considering x and y, since we only have
+    * three hydrophones
+    """
+    distance_diff = float(
+        math.sqrt(
+            math.pow(x_pos_es - param.TestParameters.SOURCE_POS_X, 2)
+            + math.pow(y_pos_es - param.TestParameters.SOURCE_POS_Y, 2)
+        )
+    )
+
+    print(f"\nThe estimated pinger position is at (x,y) = ({x_pos_es}, {y_pos_es})")
+    print(
+        f"\nThe actual pinger position is at (x,y) = ({param.TestParameters.SOURCE_POS_X}, {param.TestParameters.SOURCE_POS_Y})"
+    )
+    print(
+        f"\nThe difference between the actual position and the estimated position is {distance_diff} m"
+    )
```

Comment:
prints are not actually useful in tests.
Rather use something like
```suggestion
    assert distance_diff > tolerance
```
Which would fail the test if the difference is larger then an accepted tolerance, e.g. set `tolerance = 5  # [m]`.